multiuser/keycloak/che-multiuser-keycloak-server/src/main/java/org/eclipse/che/multiuser/keycloak/server/KeycloakEnvironmentInitalizationFilter.java

```diff
@@ -103,10 +104,17 @@ private synchronized User getOrCreateUser(String id, String email, String userna
     try {
       return userManager.getById(id);
     } catch (NotFoundException e) {
+      final UserImpl cheUser = new UserImpl(id, email, username, generate("", 12), emptyList());
       try {
-        final UserImpl cheUser = new UserImpl(id, email, username, "secret", emptyList());
         return userManager.create(cheUser, false);
-      } catch (ServerException | ConflictException ex) {
+      } catch (ConflictException ex) {
+        try {
+          cheUser.setName(generate(cheUser.getName(), 4));
+          return userManager.create(cheUser, false);
+        } catch (ServerException | ConflictException e1) {
+          throw new ServletException("Unable to create new user", e1);
```

Comment:
With such try catches this method looks unreadable. Please consider to rewrite it somehow, e.g.
```
  private synchronized User getOrCreateUser(String id, String email, String username)
      throws ServletException {
    try {
      Optional<User> user = doGetUser(id);

      if (user.isPresent()) {
        return user.get();
      }

      final UserImpl cheUser = new UserImpl(id, email, username, generate("", 12), emptyList());
      try {
        return userManager.create(cheUser, false);
      } catch (ConflictException ex) {
        cheUser.setName(generate(cheUser.getName(), 4));
        return userManager.create(cheUser, false);
      }
    } catch (ServerException | ConflictException e) {
      throw new ServletException("Unable to create new user", e);
    }
  }

  private Optional<User> doGetUser(String id) throws ServerException {
    try {
      return Optional.of(userManager.getById(id));
    } catch (NotFoundException e) {
      return Optional.empty();
    }
  }
```

Reply:
1:1 by readability imo...:)

Reply:
BTW `throw new ServletException("Unable to create new user", e);` this line is one time in my variant :smiley: